src/viam/sdk/components/camera/camera.cpp

```diff
@@ -24,6 +25,96 @@ API API::traits<Camera>::api() {
     return {kRDK, kComponent, "camera"};
 }
 
+// Appends a uint64_t value in big-endian format to a byte vector and updates the offset.
```

Comment:
Just curious, but what "format" are we following here? How did big-endian get into the picture (hah!) at all?

Reply:
Here is the context that I have: one of the reasons we are concerned with endianness is because the [Intel Realsense returns big-endian by default](https://github.com/viamrobotics/viam-camera-realsense?tab=readme-ov-file#attributes). Similarly, [the Python SDK also assumes big-endian when decoding](https://github.com/hexbabe/sean-viam-python-sdk/blob/5accda204293685954eb7382859f97c7f57b964b/src/viam/media/video.py#L43). I think @bhaney has more accurate and precise info on why we do this.

Reply:
endianness in this context is the endian-ness of a 16bit number, not of a system. Each pixel in the depth image is 2 bytes. For the two bytes, you can either read [first second] or [second first] within the byte stream. When I talk about big-endian, I mean it within this context

Reply:
calling `$xxd -g 1 test_image.viam.dep` on a test depth file, where a 20 x 10 image has each pixel value as x_location*y_location, this is what you would expect as the byte stream (with a space between every byte):
```
00000000: 44 45 50 54 48 4d 41 50 00 00 00 00 00 00 00 14  DEPTHMAP........
00000010: 00 00 00 00 00 00 00 0a 00 00 00 00 00 00 00 00  ................
00000020: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00  ................
00000030: 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00  ................
00000040: 00 00 00 01 00 02 00 03 00 04 00 05 00 06 00 07  ................
00000050: 00 08 00 09 00 0a 00 0b 00 0c 00 0d 00 0e 00 0f  ................
00000060: 00 10 00 11 00 12 00 13 00 00 00 02 00 04 00 06  ................
00000070: 00 08 00 0a 00 0c 00 0e 00 10 00 12 00 14 00 16  ................
00000080: 00 18 00 1a 00 1c 00 1e 00 20 00 22 00 24 00 26  ......... .".$.&
```
`44 45 50 54 48 4d 41 50` is the string DEPTHMAP in ascii
`00 00 00 00 00 00 00 14` says the image has a width of 20 pixels
`00 00 00 00 00 00 00 0a` says the image has a height of 10 pixels

These need to be big-endian encoded numbers too, because if instead you encoded the width as `14 00 00 00 00 00 00 00`, that would imply an image width of 1441151880758558720